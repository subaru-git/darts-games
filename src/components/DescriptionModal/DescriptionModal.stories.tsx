import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import DescriptionModal from './DescriptionModal';

export default {
  title: 'Components/DescriptionModal',
  component: DescriptionModal,
} as ComponentMeta<typeof DescriptionModal>;

const Template: ComponentStory<typeof DescriptionModal> = (args) => <DescriptionModal {...args} />;

export const Default = Template.bind({});
Default.args = {};
