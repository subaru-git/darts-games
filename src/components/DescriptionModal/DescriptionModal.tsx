import React, { FC, ReactNode } from 'react';
import {
  useDisclosure,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalCloseButton,
  ModalBody,
  useBreakpointValue,
} from '@chakra-ui/react';
import { AiOutlineInfoCircle } from 'react-icons/ai';
import IconButton from '@/atoms/IconButton';

type DescriptionModalProps = {
  header: string;
  description: ReactNode | ReactNode[];
  game?: boolean;
};

const DescriptionModal: FC<DescriptionModalProps> = ({ header, description, game = true }) => {
  const { isOpen, onToggle } = useDisclosure();
  const isMd = useBreakpointValue({ base: false, md: true });
  return (
    <div className='flex w-fit flex-nowrap items-center p-2'>
      {isMd ? (
        <IconButton aria-label='what is this game?' color='ghost' onClick={onToggle}>
          <div className='flex items-center gap-1'>
            <AiOutlineInfoCircle size={20} />
            <span>{`What is this${game ? ' game' : ''}?`}</span>
          </div>
        </IconButton>
      ) : (
        <IconButton aria-label='what is new game?' color='ghost' onClick={onToggle}>
          <div className='flex items-start'>
            <AiOutlineInfoCircle size={20} />
          </div>
        </IconButton>
      )}
      <Modal isOpen={isOpen} onClose={onToggle} size='2xl'>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader>{header}</ModalHeader>
          <ModalCloseButton />
          <ModalBody>{description}</ModalBody>
        </ModalContent>
      </Modal>
    </div>
  );
};

export default DescriptionModal;
