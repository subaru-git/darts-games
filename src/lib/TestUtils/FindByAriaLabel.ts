export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
