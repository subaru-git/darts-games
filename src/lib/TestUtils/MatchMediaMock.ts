import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
