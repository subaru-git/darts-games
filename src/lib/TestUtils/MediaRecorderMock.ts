export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
