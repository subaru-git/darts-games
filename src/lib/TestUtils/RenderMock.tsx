import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
