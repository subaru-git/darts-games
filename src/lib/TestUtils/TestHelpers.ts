import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}