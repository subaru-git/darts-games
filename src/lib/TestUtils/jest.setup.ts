import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
