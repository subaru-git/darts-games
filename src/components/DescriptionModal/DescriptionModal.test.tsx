import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import DescriptionModal from './DescriptionModal';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';

test('should rendering', () => {
  window.matchMedia = createMatchMedia(1100);
  const { container } = render(
    <ChakraProvider>
      <DescriptionModal header='' description={<></>} />
    </ChakraProvider>,
  );
  expect(container).toMatchSnapshot();
});
