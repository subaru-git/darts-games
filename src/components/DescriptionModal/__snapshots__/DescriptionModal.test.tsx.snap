// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="flex h-fit w-fit flex-nowrap items-start p-2"
  >
    <button
      aria-label="what is this game?"
      class="text-gray-600 bg-transparent hover:text-gray-400 p-0 focus:ring-0 disabled:opacity-50 m-auto inline-flex items-center rounded-full p-2.5 text-center text-sm font-medium focus:outline-none disabled:cursor-not-allowed"
      type="button"
    >
      <div
        class="flex items-center gap-1"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
          />
        </svg>
        <span>
          What is this game?
        </span>
      </div>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
